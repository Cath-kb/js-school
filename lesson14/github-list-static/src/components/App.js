import React from 'react';
import UserList from './UserList';

class App extends React.Component {
  render() {
    const {users} = this.props;
    const title = 'Github User List Static';

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        <UserList users={users} />
      </div>
    );
  }
}

export default App;
