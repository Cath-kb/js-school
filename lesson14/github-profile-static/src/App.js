import React from 'react';
import Header from './Layout/Header';
import Content from './Layout/Content';
import User from './User';

class App extends React.Component {
  render() {
    const {user} = this.props;
    const title = 'Github User Profile Static';

    return (
      <div className="container my-5">
        <Header title={title} />
        <Content>
          <User user={user} />
        </Content>
      </div>
    );
  }
}

export default App;
