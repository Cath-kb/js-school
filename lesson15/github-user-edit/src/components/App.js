import React from 'react';

import User from './User';
import Form from './User/Form';

import { getGithubUser, updateGithubUser } from '../model/api';

import { MODE_VIEW, MODE_EDIT, MODE_LOAD } from '../consts';
import { GITHUB_USERNAME } from '../configs';

class App extends React.Component {
  state = {
    mode: MODE_LOAD,
    user: null,
  };

  toggleMode = () => {
    this.setState({
      mode: this.state.mode === MODE_VIEW ? MODE_EDIT : MODE_VIEW,
    });
  };

  updateUser = (person) => {
    this.setState({
      mode: MODE_LOAD,
    });
    updateGithubUser(person)
      .then(user => this.setState({
        mode: MODE_VIEW,
        user,
      }));
  };

  isMode(mode) {
    return this.state.mode === mode;
  }

  componentDidMount() {
    getGithubUser(GITHUB_USERNAME)
      .then(user => this.setState({
        mode: MODE_VIEW,
        user,
      }));
  }

  render() {
    const { user } = this.state;

    const title = `Github User Profile Fetching and Updating ${this.isMode(MODE_EDIT) ? '(Edit)': ''}`;

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        {this.isMode(MODE_VIEW) && <User user={user} onClick={this.toggleMode} />}
        {this.isMode(MODE_EDIT) && <Form initialUser={user} onReset={this.toggleMode} onSubmit={this.updateUser} />}
        {this.isMode(MODE_LOAD) && <h2 className="my-5 text-center">Loading&hellip;</h2>}
      </div>
    );
  }
}

export default App;
