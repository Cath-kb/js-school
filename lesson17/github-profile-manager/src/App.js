import React from 'react';
import { Switch, Route, Redirect } from 'react-router';

import Profile from "./components/Profile";
import EditForm from "./components/EditForm";

import { getGithubUser } from './model/api';

class App extends React.Component {
  state = {
    isLoading: true,
    user: null,
  };

  getUser(user) {
    getGithubUser(user)
      .then(user => this.setState({
        isLoading: false,
        user,
      }));
  }

  handleSuccess = user => {
    this.setState({
      user,
    });
  };

  componentDidMount() {
    const { user_login } = this.props.match.params;
    this.getUser(user_login);
  }

  componentDidUpdate(prevProps) {
    const { user_login } = this.props.match.params;

    if (user_login !== prevProps.match.params.user_login) {
      this.getUser(user_login);
    }
  }

  render() {
    const title = 'Github Profile Manager React App';

    const { user_login } = this.props.match.params;
    const { user, isLoading } = this.state;

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        {
          isLoading ? <h2 className="my-5 text-center">Loading&hellip;</h2> :
          <Switch>
            <Route path="/:user_login/edit" render={props => <EditForm initialUser={user} {...props} onSuccess={this.handleSuccess} />} />
            <Route path="/:user_login" render={props => <Profile user={user} {...props} />} />
            <Route render={() => <Redirect to={`/${user_login}`} />} />
          </Switch>
        }
      </div>
    );
  }
}

export default App;
