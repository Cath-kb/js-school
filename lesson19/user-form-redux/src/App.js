import React, { Component } from 'react';
import UserForm from './UserForm';
import UserInfo from './UserInfo';

// DO NOT Modify this file

class App extends Component {
  render() {
    return (
      <div className="container">
        <header className="header">
          <h1>User Form</h1>
        </header>
        <UserForm />
        <UserInfo />
      </div>
    );
  }
}

export default App;
