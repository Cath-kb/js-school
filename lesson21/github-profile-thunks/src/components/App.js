import React from 'react';
import { connect } from 'react-redux';

import { load } from '../store';

import { Header, Content, Loading, Error } from './Layout';
import User from './User';

class App extends React.Component {
  componentDidMount() {
    const { user_login } = this.props.match.params;
    this.props.loadUser(user_login);
  }

  render() {
    const { isLoading, user, error } = this.props;
    const title = 'Github User Profile Redux Thunks';

    return (
      <div className="container my-5">
        <Header title={title} />
        {isLoading ?
          <Loading /> :
          user ?
          <Content>
            <User user={user} />
          </Content> :
          error ?
          <Error error={error} /> :
          null
        }
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.isLoading,
  user: state.user,
  error: state.error,
});

const mapDispatchToProps = dispatch => ({
  loadUser: user_login => dispatch(load(user_login)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
