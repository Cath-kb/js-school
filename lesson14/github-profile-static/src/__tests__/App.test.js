import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './../App';

const app = {
  title: 'Test title',
  user: {
    avatar: {
      url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
    },
    person: {
      name: 'John Doe',
      login: 'joe',
      company: '',
      location: 'Aldebaran',
    },
    statistics: {
      repos: 5,
      following: 0,
      followers: 5,
    }
  }
};

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App user={app.user} title={app.title} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<App user={app.user} title={app.title} />).toJSON();
    expect(html).toMatchSnapshot();
  });
});
