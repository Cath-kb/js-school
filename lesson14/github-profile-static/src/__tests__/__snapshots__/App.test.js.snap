// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be rendered without errors 1`] = `
<div
  className="container my-5"
>
  <header
    className="header"
  >
    <h1
      className="title text-center mb-4"
    >
      Github User Profile Static
    </h1>
  </header>
  <div>
    <div
      className="row"
    >
      <div
        className="col-md-6  mb-4"
      >
        <img
          alt="User Avatar"
          className="img-fluid mx-auto d-block rounded"
          src="https://avatars3.githubusercontent.com/u/2164575?v=4"
        />
      </div>
      <div
        className="col-md-6"
      >
        <div>
          <h2>
            John Doe
          </h2>
          <h3>
            joe
          </h3>
          
          <p>
            Aldebaran
          </p>
        </div>
        <div>
          <h4>
            <strong>
              Statistics:
            </strong>
          </h4>
          <ul
            className="list-group"
          >
            <li
              className="list-group-item d-flex justify-content-between align-items-center"
            >
              Repositories
              : 
              <span
                className="badge badge-primary badge-pill"
              >
                5
              </span>
            </li>
            <li
              className="list-group-item d-flex justify-content-between align-items-center"
            >
              Following
              : 
              <span
                className="badge badge-primary badge-pill"
              >
                0
              </span>
            </li>
            <li
              className="list-group-item d-flex justify-content-between align-items-center"
            >
              Followers
              : 
              <span
                className="badge badge-primary badge-pill"
              >
                5
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
